chapters/ch10.md

```diff
@@ -1,23 +1,25 @@
 1. Start off with some ​**idea​** on how to build the system.
 ->
-
+Bắt đầu với một vài ý tưởng xây dựng hệ thống.
 2. Implement the idea in ​**code​**.
 ->
-
+Hiện thực hóa ý tưởng dưới dạng code.
 3. Carry out an ​**experiment​** which tells me how well the idea worked. (Usually my first few ideas don’t work!) Based on these learnings, go back to generate more ideas, and keep on iterating.
 ->
+Tiến hành một **thí nghiệm** cho đo tính khả thi của ý tưởng. (Thường thì một số ý tưởng đầu tiên sẽ không khả thi!) Từ những kết quả đó, chúng ta quay lại thử nghiệm thêm những ý tưởng mới và cứ thế lặp lại
 
 ![img](../imgs/C10_01.png)
 
 This is an iterative process. The faster you can go round this loop, the faster you will make progress. This is why having dev/test sets and a metric are important: Each time you try an idea, measuring your idea’s performance on the dev set lets you quickly decide if you’re heading in the right direction.
 ->
+Đây là một quá trình lặp đi lặp lại. Hoàn thiện vòng lặp càng nhanh, thì càng sớm cải thiện kết quả. Đó là lý do tại sao có tập phát triển/thử nghiệm và một phép đo là rất quan trọng: Việc đánh giá chất lượng của mỗi ý tưởng trên tập phát triển giúp xác định liệu chúng ta có đi đúng hướng.
 
 In contrast, suppose you don’t have a specific dev set and metric. So each time your team develops a new cat classifier, you have to incorporate it into your app, and play with the app for a few hours to get a sense of whether the new classifier is an improvement. This would be incredibly slow! Also, if your team improves the classifier’s accuracy from 95.0% to 95.1%, you might not be able to detect that 0.1% improvement from playing with the app. Yet a lot of progress in your system will be made by gradually accumulating dozens of these 0.1% improvements. Having a dev set and metric allows you to very quickly detect which ideas are successfully giving you small (or large) improvements, and therefore lets you quickly decide what ideas to keep refining, and which ones to discard.
 ->
-
+Ngược lại, giả sử bạn không có một tập phát triển và phép đo cụ thể. Như vậy mỗi khi cộng sự của bạn phát triển một bộ phân loại mèo, bạn sẽ phải tích hợp nó vào ứng dụng, và ngồi vọc ứng dụng đó một vài tiếng để biết rằng bộ phân loại mới có thật sự tốt hay không. Nó sẽ cực kì chậm! Ngoài ra, nếu cộng sự của bạn cải thiện độ chính xác từ 95.0% lên 95.1%, bạn sẽ không thể phát hiện sự cải thiện 0.1% đó chỉ qua việc ngồi vọc ứng dụng. Còn nữa, sẽ có rất nhiều tiến triển trên hệ thống của bạn thông qua việc tích lũy những sự cải thiện 0.1% này. Có một tập phát triển và phép đo cho phép bạn nhanh chóng phát hiện ra ý tưởng nào sẽ thật sự cho bạn những cải thiện nhỏ (hoặc lớn), và từ đó bạn có thể quyết định những ý tưởng nào cần hoàn thiện thêm, và những ý tưởng nào cần loại bỏ.
```

Comment:
Em có thể tách cả đoạn lớn này thành 2-3 đoạn nhỏ cho tiện review.

- team --> dịch là nhóm cho đồng thuận với các phần trước. 
- "của bạn phát triển một bộ phân loại mèo," --> thiếu từ "new"
- "vọc" --> chọn từ mang tính academic hơn tí
- "để biết rằng bộ phân loại mới có thật sự tốt hay không." --> Câu này dịch không sát nghĩa gốc. Ý là kiểm tra xem có cải thiện (so với cái cũ) hay không. 
--> để kiểm tra liệu bộ phân loại mới có cải thiện hay không. 
- "Nó" ở đây là "quá trình này", cần làm rõ nghĩa. "would be" dịch nghĩa có thể thì đúng hơn. Vì quá trình này có thể nhanh chứ không phải luôn luôn chậm. 
> Nó sẽ cực kì chậm!
- Also --> đồng thời, sẽ hợp hơn là "ngoài ra". 
> Ngoài ra, nếu cộng sự của bạn cải thiện độ chính xác từ 95.0% lên 95.1%, bạn sẽ không thể phát hiện sự cải thiện 0.1% đó chỉ qua việc ngồi vọc ứng dụng. 

Mình xin viết lại cho thuận hơn
> Đồng thời, nhóm của bạn sẽ rất khó nhận ra sự khác biệt nếu chỉ cải thiện độ chính xác từ 95.0% lên 95.1%. 

- Câu này khá khó dịch. Mình nêu tạm ý của mình. Vẫn chưa ưng lắm. 
> Còn nữa, sẽ có rất nhiều tiến triển trên hệ thống của bạn thông qua việc tích lũy những sự cải thiện 0.1% này.

--> "Và hệ thống sau cùng là tích lũy của rất nhiểu bước cải thiện nhỏ 0.1%"

- không có từ "thực sự"
- cho bạn --> đem lại 
- cải thiện --> cải tiến
- -->  những ý tưởng cần cải thiện thêm hoặc loại bỏ. 
> Có một tập phát triển và phép đo cho phép bạn nhanh chóng phát hiện ra ý tưởng nào sẽ thật sự cho bạn những cải thiện nhỏ (hoặc lớn), và từ đó bạn có thể quyết định những ý tưởng nào cần hoàn thiện thêm, và những ý tưởng nào cần loại bỏ.